github/utils.go

```diff
@@ -40,15 +55,61 @@ func connect(ctx context.Context, d *plugin.QueryData) *github.Client {
 		baseURL = *githubConfig.BaseURL
 	}
 
-	if token == "" {
-		panic("'token' must be set in the connection configuration. Edit your connection configuration file and then restart Steampipe")
+	//// Github App authentication.
+	if githubConfig.AppId != nil {
+		githubAppId = *githubConfig.AppId
+	}
+	if githubConfig.InstallationId != nil {
+		githubInstallationId = *githubConfig.InstallationId
+	}
+	if githubConfig.PrivateKey != nil {
+		privateKeyPath = *githubConfig.PrivateKey
+	}
+
+	if token == "" && (githubAppId == "" || githubInstallationId == "" || privateKeyPath == "") {
+		panic("'token' or 'app_id', 'installation_id' and 'private_key' or `app_installation_access_token` must be set in the connection configuration. Edit your connection configuration file and then restart Steampipe")
+	}
+
+	// Return error for unsupported token by prefix
+	if token != "" && !strings.HasPrefix(token, "ghs_") && !strings.HasPrefix(token, "ghp_") {
```

Comment:
Do we need to add this check? Do we have any other tokens that can be passed as a config argument?

Reply:
Yes, we need this check, we have other tokens that can be passed as token config arguments.

Eventual support of other token formats would be good. We tried fine-grained tokens but ran into some issues, so that requires more research, but we deprioritized it for the time being.

For more information about the supported token format, please see: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-authentication-to-github#githubs-token-formats